bin/i18n-map-extractor.js

```diff
@@ -0,0 +1,130 @@
+const recast = require( 'recast' );
+const { forEach, reduce, includes, has, isEmpty, isFunction } = require( 'lodash' );
+const { writeFileSync } = require( 'fs' );
+const NormalModule = require( 'webpack/lib/NormalModule' );
+
+class wpi18nExtractor {
+	constructor( options ) {
+		const DEFAULT_FUNCTIONS = [
+			'__',
+			'_n',
+			'_x',
+			'_nx',
+			'sprintf',
+		];
+		this.options = options || {};
+		this.options.filename = this.options.filename || 'translation-map.json';
+		this.translationMap = {};
+		this.functionNames = this.options.functionNames || DEFAULT_FUNCTIONS;
+	}
+
+	extractStringFromFunctionCall( args, functionName ) {
+		const strings = [];
+		switch ( functionName ) {
+			case '__':
+			case 'sprintf':
+			case '_n':
+				strings.push( args[ 0 ].value );
+				break;
+			case '_x':
+				strings.push( args[ 1 ].value + '\u0004' + args[ 0 ].value );
+				break;
+			case '_nx':
+				strings.push( args[ 3 ].value + '\u0004' + args[ 0 ].value );
```

Comment:
I had noticed that the msgID format for `_x` and `_nx` strings is `{context}\u0004{singular_string}` so I corrected that.  This took care of a few strings that weren't translating in my tests.